Safewaan/HonkedIn
# Implemented My Resources page with search, filtering, edit, and delete functionalities

## Trello ticket link

[As a user, I want to be able to view my resources.](https://trello.com/c/URj8ZmVj )
[As a user, I want to be able to edit a resource I posted.](https://trello.com/c/vTryEfaA )
[As a user, I want to delete resources I posted.](https://trello.com/c/zE6c56SH )



## Implementation description
* Created the My Resources page which lists all resources created by the user 
* Implemented Search function which allows user to search for resources based on name e
* Implemented filtering based on the Media tags 
* Added edit resource functionality which allows users to edit the name, link, and or media tags for a given resource 
* Added delete resource functionality which allows users to delete a given resource. This delete function has a confirmation pop-up.
* Added My Resources page to the Navigation Bar 
* Fixed small error on the forums page so the constant variable holding the filtering forumTag is different than that which is used to edit the forumTag



## Steps to test
1. Make sure you can access the My Resources page 
2. Ensure that you can see the list of all resources that you've created that are stored in the db 
3. Test that the search function works, and that a user can search for a list of resources that have names that match the search term 
4. Check that if the search bar and filtering is empty, the whole list should be generated 
5. Check that clicking clear search button clears the search bar and re-renders the entire list (with filtering also empty)
6. Test that the filtering works based on media tag 
7. Check that you can remove the filter 
8. Ensure that a user can click the "Edit resource" button 
9. Make sure that a pop-up appears with the given resource's current values pre-populated in the fields 
10. Make sure you can enter new values in each of the fields 
11. Ensure that if you leave either the name, or link field empty, you will receive and error and be unable to edit the resource 
12. Ensure that you can leave the media tag field empty 
13. Make sure that a successful edit toast message pops up once you click "Edit resource" 
14. Ensure that clicking "Close" on the pop-up, closes it and does not change the resource values 
15. Next make sure you can click the Delete resource button that opens a confirmation pop-up 
16. Ensure that clicking yes deletes the given resource 
17. Ensure that clicking no closes the pop-up and does not delete the resource



## What should reviewers focus on?
* Make sure the functionality matches that of the other pages on our platform
